xarray/core/indexing.py

```diff
@@ -1877,23 +1896,30 @@ class PandasMultiIndexingAdapter(PandasIndexingAdapter):
     __slots__ = ("_dtype", "adapter", "array", "level")
 
     array: pd.MultiIndex
-    _dtype: np.dtype
+    _dtype: np.dtype | pd.api.extensions.ExtensionDtype
     level: str | None
 
     def __init__(
         self,
         array: pd.MultiIndex,
-        dtype: DTypeLike = None,
+        dtype: DTypeLike | pd.api.extensions.ExtensionDtype | None = None,
         level: str | None = None,
     ):
         super().__init__(array, dtype)
         self.level = level
 
     def __array__(
-        self, dtype: np.typing.DTypeLike = None, /, *, copy: bool | None = None
+        self,
+        dtype: DTypeLike | pd.api.extensions.ExtensionDtype | None = None,
+        /,
+        *,
+        copy: bool | None = None,
     ) -> np.ndarray:
         if dtype is None:
             dtype = self.dtype
+        if pd.api.types.is_extension_array_dtype(dtype):
```

Comment:
is this needed? Why would someone call `np.array` with an extension dtype, and then expect it to get translated to a numpy dtype?

Reply:
This is mostly for internal usage, otherwise I wouldn't have added it.  I can delete the line and then see what happens, and then comment.

Reply:
@dcherian This class is basically an internal adapter so anything that asks for its data in numpy form.  Things like `repr`, subtraction, and calling `.values` on an `xarray` object are a few examples

Reply:
Hh, good point, yes, it's unnecessary.  Fixed.